bigquery_etl/shredder/bug1751979.py

```diff
@@ -370,3 +370,134 @@
+  sanitize_scalar(search_content_webextension_sum) AS search_content_webextension_sum,
+  sanitize_scalar(search_content_system_sum) AS search_content_system_sum,
+  sanitize_scalar(search_content_tabhistory_sum) AS search_content_tabhistory_sum,
+  sanitize_scalar(search_content_reload_sum) AS search_content_reload_sum,
+  sanitize_scalar(search_content_unknown_sum) AS search_content_unknown_sum
+)
+"""
+
+BUG_1751979_SEARCH_CLIENTS_DAILY_V8_UDFS = r"""
+CREATE TEMP FUNCTION sanitize_engine_source(
+  engine STRING, `source` STRING
+) AS ((
+    WITH base AS (
+      SELECT CONCAT(engine, '.', `source`) AS key
+    ),
+    parsed AS (
+      SELECT
+        *,
+        REGEXP_EXTRACT(key, "([^.]+[.]in-content[:.][^:]+:).*") AS prefix,
+        REGEXP_EXTRACT(key, "[^.]+[.]in-content[:.][^:]+:(.*)") AS code,
+      FROM
+        base
+    ),
+    scrubbed AS (
+      SELECT
+          IF(
+            prefix IS NULL
+            OR code IN (
+              "none",
+              "other",
+              "hz",
+              "h_",
+              "MOZ2",
+              "MOZ4",
+              "MOZ5",
+              "MOZA",
+              "MOZB",
+              "MOZD",
+              "MOZE",
+              "MOZI",
+              "MOZM",
+              "MOZO",
+              "MOZT",
+              "MOZW",
+              "MOZSL01",
+              "MOZSL02",
+              "MOZSL03",
+              "firefox-a",
+              "firefox-b",
+              "firefox-b-1",
+              "firefox-b-ab",
+              "firefox-b-1-ab",
+              "firefox-b-d",
+              "firefox-b-1-d",
+              "firefox-b-e",
+              "firefox-b-1-e",
+              "firefox-b-m",
+              "firefox-b-1-m",
+              "firefox-b-o",
+              "firefox-b-1-o",
+              "firefox-b-lm",
+              "firefox-b-1-lm",
+              "firefox-b-lg",
+              "firefox-b-huawei-h1611",
+              "firefox-b-is-oem1",
+              "firefox-b-oem1",
+              "firefox-b-oem2",
+              "firefox-b-tinno",
+              "firefox-b-pn-wt",
+              "firefox-b-pn-wt-us",
+              "ubuntu",
+              "ffab",
+              "ffcm",
+              "ffhp",
+              "ffip",
+              "ffit",
+              "ffnt",
+              "ffocus",
+              "ffos",
+              "ffsb",
+              "fpas",
+              "fpsa",
+              "ftas",
+              "ftsa",
+              "newext",
+              "monline_dg",
+              "monline_3_dg",
+              "monline_4_dg",
+              "monline_7_dg"
+            ),
+            key,
+            CONCAT(
+              prefix,
+              "other.scrubbed.",
+              -- Add a hash of code to avoid multiple rows ending up with the same
+              -- (client_id, sanitized_engine, sanitized_source) value, violating
+              -- the table's contract.
+              SUBSTR(TO_HEX(SHA256(code)), 1, 4))
```

Comment:
* How confident are we that a 4-character hash will be enough to not have any collisions? (65k isn't that big of a hash space)
* This won't entirely avoid some potential adverse effects of having a large number of `source` values (these "other.scrubbed.xxxx" values won't easily aggregate together, and having thousands of such values could overwhelm users if they're someplace like in a BI tool filter select list).
* `SUBSTR(value, 1, 4)` could be more succinctly/clearly done with `LEFT(value, 4)`.

Reply:
I've updated to use LEFT and take 8 characters rather than 4.

> these "other.scrubbed.xxxx" values won't easily aggregate together, and having thousands of such values could overwhelm users if they're someplace like in a BI tool filter select list

Agreed, but the situation will be no worse than it is today with unscrubbed values. We've properly recreated the data from 2019-11-22 and later from `clients_daily`, so this logic will only be relevant for older history. We can do a full recreation in a month after we're sure the entirety of `clients_daily` history is sanitized.